libpromises/evalfunction.c

```diff
@@ -1188,6 +1188,95 @@ static FnCallResult FnCallGetGid(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const P
 #endif
 }
 
+/*********************************************************************/ 
+
+static FnCallResult FnCallUserInGroup(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+{
+    assert(fp != NULL);
+#ifdef _WIN32
+    Log(LOG_LEVEL_ERR, "Function '%s' is POSIX specific", fp->name);
+    return FnFailure();  
+#else
+    assert(finalargs != NULL);
+
+    const char *user_name = RlistScalarValue(finalargs);
+    const char *group_name = RlistScalarValue(finalargs->next);
+
+    StringSet *users = StringSetNew();
+    int ret;
+
+    // check secondary group
```

Comment:
Here you could mention that this is what we expect the function to be used for the most, which is why we are checking it first (before primary group).